src/foam/lib/PermissionedPropertyPredicate.js

```diff
@@ -0,0 +1,40 @@
+foam.CLASS({
+    package: 'foam.lib',
+    name: 'PermissionedPropertyPredicate',
+    implements: [ 'foam.lib.PropertyPredicate'],
+    javaImports: [
+      'foam.nanos.auth.AuthService'
+    ],
+    
+    methods: [
+      {
+        name: 'propertyPredicateCheck',
```

Comment:
Same here. Delete the type and args and let FOAM fill them in by you implementing the interface.